coderd/database/dbauthz/dbauthz.go

```diff
@@ -3547,6 +3561,13 @@ func (q *querier) InsertWorkspaceAppStats(ctx context.Context, arg database.Inse
 	return q.db.InsertWorkspaceAppStats(ctx, arg)
 }
 
+func (q *querier) InsertWorkspaceAppStatus(ctx context.Context, arg database.InsertWorkspaceAppStatusParams) (database.WorkspaceAppStatus, error) {
+	if err := q.authorizeContext(ctx, policy.ActionCreate, rbac.ResourceSystem); err != nil {
```

Comment:
As above, we need to stop the proliferation of ResourceSystem. We only want folks who can update the related workspace agent to be able to insert related app statuses.

Reply:
I'm confused, because this is just from the agent - I'm really just following inserting apps and such semantics.

Reply:
Yeah, after looking a little closer I see now that it's unfortunately a fairly well-established pattern. This may need to be addressed separately as part of a larger follow-up.

Reply:
For posterity, here is the 'catch-all' follow-up issue: https://github.com/coder/coder/issues/13315